### DISCUSSION TITLE
Container queries plugin - just adding `@container` changes behaviour of the container?

### DISCUSSION THREAD

**Discussion by silverdr** [tailwindlabs/tailwindcss#14230] [Help] [answered]
I am wondering if that's expected behaviour that only adding `@container` to an element can change the way it is rendered?

My case is that I have resizeable "window" div, which has some content inside one of the inner `div`s. It opens and renders nicely as expected, having – as expected – all content visible in one line. Now I add `@container` to the `div`, and… it no longer "notices" its content and as a result the width of the whole "window is limited by its footer content rather than the main div's one.

A bug? Or what am I doing wrong?

![Screenshot 2024-08-21 at 13 42 54](https://github.com/user-attachments/assets/3caeadfe-9b13-4944-a9a6-2157e018db85)
![Screenshot 2024-08-21 at 13 43 14](https://github.com/user-attachments/assets/f0133daf-8c99-48ff-8d87-b2db729dceeb)

**Comment by wongjn** [comment, marked as answer]
This is expected behavior. `@container` applies `container-type: inline-size`, of which, according to the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/container-type#inline-size) (emphasis mine):

> `inline-size`  
> Establishes a query container for dimensional queries on the [inline axis](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_logical_properties_and_values/Basic_concepts_of_logical_properties_and_values#block_and_inline_dimensions) of the container. Applies layout, style, and inline-size containment to the element.
> Inline size containment is applied to the element. **The inline size of the element can be computed in isolation, ignoring the child elements**.

**Reply by silverdr** [reply]
Thank you for the answer and explanations!
